sorting/count_inversions.cpp

```diff
@@ -0,0 +1,267 @@
+/**
+ * @file
+ * @brief Counting Inversions using [Merge
+ Sort](https://en.wikipedia.org/wiki/Merge_sort)
+ *
+ * @details
+ * Program to count the number of inversions in an array
+ * using merge-sort.
+ *
+ * The count of inversions help to determine how close the array
+ * is to being sorted in ASCENDING order.
+ *
+ * two elements a[i] and a[j] form an inversion if a[i] > a[j] and i < j
+ *
+ * Time Complexity --> O(n.log n)
+ * Space Complexity --> O(n) ; additional arrat temp[1..n]
```

Comment:
```suggestion

 * Space Complexity --> O(n) ; additional array temp[1..n]
```

Reply:
Adding an extra space because the documentation currently looks like this
<img width="1503" alt="Screenshot 2020-11-24 at 12 49 15" src="https://user-images.githubusercontent.com/21105489/100096113-7977f500-2e53-11eb-842a-84c0d8aeb78b.png">